src/components/policy/src/policy/src/access_remote_impl.cc

```diff
@@ -405,12 +504,55 @@ bool AccessRemoteImpl::CheckPTUUpdatesChange(
     return false;
   }
 
-  if (new_consent.is_initialized() && old_consent.is_initialized())
-    return *new_consent != *old_consent;
+  bool result = false;
+  if (new_consent.is_initialized() && old_consent.is_initialized()) {
+    result = (*new_consent != *old_consent);
+  } else {
+    bool not_changed_consent1 = !new_consent.is_initialized() && *old_consent;
+    bool not_changed_consent2 = !old_consent.is_initialized() && *new_consent;
+
+    result = !(not_changed_consent1 || not_changed_consent2);
+  }
 
-  bool not_changed_consent1 = !new_consent.is_initialized() && *old_consent;
-  bool not_changed_consent2 = !old_consent.is_initialized() && *new_consent;
-  return !(not_changed_consent1 || not_changed_consent2);
+  if (result) {
+    listener_->OnRemoteAllowedChanged(result);
+  }
+
+  return result;
+}
+
+void AccessRemoteImpl::CheckPTUZonesChange(
+    const utils::SharedPtr<policy_table::Table> pt_update,
+    const utils::SharedPtr<policy_table::Table> snapshot) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  rpc::Optional<Equipment>& old_equipment =
+    snapshot->policy_table.module_config.equipment;
+  rpc::Optional<Equipment>& new_equipment =
+    pt_update->policy_table.module_config.equipment;
+
+  if (!old_equipment.is_initialized() || !new_equipment.is_initialized()) {
+    LOG4CXX_DEBUG(logger_, "No need to reset access, equipment is not inited");
+    return;
+  }
+  // TODO(KK): change to erasing only affected by PTU changes permissions
+  // when described in requirements
+  acl_.clear();
+}
+
+void AccessRemoteImpl::CheckPTUGroupsChange(
```

Comment:
Different between AccessRemoteImpl::CheckPTUGroupsChange and PM::CheckPermissionsChanges is only the functor which you use to check groups changed.